src/function.ts

```diff
@@ -198,54 +198,12 @@ export function flow<A extends ReadonlyArray<unknown>, B, C, D, E, F, G, H, I, J
   hi: (h: H) => I,
   ij: (i: I) => J
 ): (...a: A) => J
-export function flow(
-  ab: Function,
-  bc?: Function,
-  cd?: Function,
-  de?: Function,
-  ef?: Function,
-  fg?: Function,
-  gh?: Function,
-  hi?: Function,
-  ij?: Function
-): unknown {
-  switch (arguments.length) {
-    case 1:
-      return ab
-    case 2:
-      return function (this: unknown) {
-        return bc!(ab.apply(this, arguments))
+export function flow(...fns: ReadonlyArray<Function>): unknown {
+  return fns.length === 0 || fns.length > 9
+    ? undefined
+    : function (this: unknown, ...args: ReadonlyArray<unknown>) {
+        return fns.slice(1).reduce((prev, fn) => fn(prev), fns[0].apply(this, args))
```

Comment:
Something to consider: we used something like this in [`pipe-ts`](https://github.com/unsplash/pipe-ts/blob/920d878046c0e7b719ee9c154b97bff1e2868215/src/index.ts#L68) but the problem we found was that it added a lot of noise to stack traces. Example:

```
Error: Expected Some but got None
  at onNone(./shared/facades/option/operators.ts:31:15)
  at apply(./node_modules/fp-ts/es6/Option.js:144:49)

 ---------  PIPE STARTS -----------------
  at ? (./node_modules/pipe-ts/target/index.js:15:23)
  at Array.reduce(<anonymous>)
  at pipe(./node_modules/pipe-ts/target/index.js:13:20)
  at e.pipeWith(./node_modules/pipe-ts/target/index.js:25:12)
 ---------  PIPE END --------------------
  
  at apply(./shared/facades/option/operators.ts:27:3)
  at ? (./node_modules/pipe-ts/target/index.js:15:23)
  at Array.reduce(<anonymous>)
```

When you use `pipe`/`flow` everywhere this becomes really important, especially if you consider that by default Node limits stack traces to 10 lines.

The existing `switch` implementation doesn't add any uneccessary noise to the stack trace.

We actually switched from `pipe-ts` to using `fp-ts` `pipe`/`flow` for this very reason.

Reply:
@OliverJAsh what about the alternative implementation I'm using in the perf folder?

```ts
export function flow(...fns: ReadonlyArray<Function>): Function | undefined {
  const len = fns.length - 1
  return len > 8
    ? undefined
    : function (this: any, ...x: ReadonlyArray<any>) {
        let y = fns[0].apply(this, x)
        for (let i = 1; i <= len; i++) {
          y = fns[i].call(this, y)
        }
        return y
      }
}
```

the stack trace seems the same as the existing implementation

Reply:
Yes that should be fine.